tracer/build/_build/Build.Steps.cs

```diff
@@ -652,44 +652,39 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
+            var testProjects = Solution.GetProjects("*.Tests")
+                                       .Where(p => p.Path.ToString().EndsWith(".csproj")); // exclude native test projects
+            foreach(var projectName in testProjects)
             {
+                Logger.Information("Copying native files for project {ProjectName}", projectName);
                 var project = Solution.GetProject(projectName);
-                var testDir = project.Directory;
+                var testDir = project!.Directory;
                 var frameworks = project.GetTargetFrameworks();
-                var testBinFolder = testDir / "bin" / BuildConfiguration;
 
-                if (IsWin)
+                if (Framework is not null)
                 {
-                    foreach (var framework in frameworks)
-                    {
-                        var source = MonitoringHomeDirectory / $"win-{TargetPlatform}" / "datadog_profiling_ffi.dll";
-                        var dest = testBinFolder / framework / "LibDatadog.dll";
-                        CopyFile(source, dest, FileExistsPolicy.Overwrite);
-                    }
+                    frameworks = frameworks.Where(x=> x == Framework).ToList();
                 }
-                else if (IsLinux)
+
+                var testBinFolder = testDir / "bin" / BuildConfiguration;
+
+                var (ext, source) = Platform switch
                 {
-                    var (arch, ext) = GetUnixArchitectureAndExtension();
-                    var source = MonitoringHomeDirectory / arch / $"libdatadog_profiling.{ext}";
-                    foreach (var framework in frameworks)
-                    {
-                        var dest = testBinFolder / framework / $"LibDatadog.{ext}";
-                        CopyFile(source, dest, FileExistsPolicy.Overwrite);
-                    }
-                }
-                else if (IsOsx)
+                    PlatformFamily.Windows => ("dll", MonitoringHomeDirectory / $"win-{TargetPlatform}" / "datadog_profiling_ffi.dll"),
+                    PlatformFamily.Linux => ("so", MonitoringHomeDirectory / GetUnixArchitectureAndExtension().Arch / "libdatadog_profiling.so"),
+                    PlatformFamily.OSX => ("dylib", MonitoringHomeDirectory / "osx" / $"libdatadog_profiling.dylib"),
+                    _ => throw new NotSupportedException($"Unsupported platform: {Platform}")
+                };
+
+                foreach (var framework in frameworks)
                 {
-                    var source = MonitoringHomeDirectory/ "osx" / $"libdatadog_profiling.dylib";
-                    foreach (var framework in frameworks)
-                    {
-                        var dest = testBinFolder / framework / $"LibDatadog.dylib";
-                        CopyFile(source, dest, FileExistsPolicy.Overwrite);
-                    }
+                    var dest = testBinFolder / framework / $"LibDatadog.{ext}";
```

Comment:
I like the approach but I have a question.

As we are looping through `frameworks` twice (once `frameworks.Where(x=> x == Framework)` and another time with this foreach) and we aren't using it between L673 and L684,
Wouldn't it be simpler to move the `where` predicate condition (`x == Framework` that is) and inverse it:
```suggestion
                    if  (framework != Framework) 
                    {
                         continue;
                    }
                    var dest = testBinFolder / framework / $"LibDatadog.{ext}";
```

now maybe there is something I'm missing  😄

Reply:
Yeah, we could, but we would also need to add the `Framework is not null` case in there, so I think it's prob simpler as is tbh.

Reply:
true I missed that sorry. 
Well in this case maybe we could simply do that 😁:
```suggestion
                    if  (Framework is not null && framework != Framework) 
                    {
                         continue;
                    }
                    var dest = testBinFolder / framework / $"LibDatadog.{ext}";
```
therefore if `Framework` is null, we use the element as is. otherwise we compare it with `Framework`'s value and skip if they are different.

I don't think it's more complicated.

*I know I don't always suggest the prettiest code** 😄

Reply:
One other thing is that the current code reads as something such as "get the frameworks that we care about, then process them" whereas the more optimized one reads as "process all of the frameworks, then determine whether we care about the framework"

It's really just a readability thing IMO, it may be more performant but I don't think that matters too much for this code in this case.

Reply:
I do see your point, here. 
I personally read the latter as "Go through all the frameworks and only process the one we care about". 
But if you all find it more difficult to read, go for the former as It's for the build so I think readability is more important still